File: Easy/20BubbleSort.js
function bubbleSort(array) {
  // Write your code here.
  let isSwap = false;
  let counter = 1;
  while (!isSwap) {
    isSwap = true;
    for (let i = 0; i < array.length + counter; i++) {
      if (array[i] > array[i + 1]) {
        swap(i, i + 1, array);
        isSwap = false;
      }
    }
    counter++;
  }
  return array;
}
function swap(i, j, array) {
  let temp = array[i];
  array[i] = array[j];
  array[j] = temp;
}

console.log(bubbleSort([8, 5, 2, 9, 5, 6, 3, 1, 10]));

File: Easy/20BubbleSort.py
def bubbleSort(array):
    # Write your code here.
    counter = 1
    isSwap = False
    while not isSwap:
        isSwap = True
        for i in range(len(array)-counter):
            if array[i] > array[i+1]:
                swap(i, i+1, array)
                isSwap = False
        counter += 1
    return array


def swap(i, j, array):
    array[i], array[j] = array[j], array[i]


print(bubbleSort([8, 5, 2, 9, 5, 6, 3, 1, 10]))
